### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by kalpanaTech**
Hi Aman, 
Please find the competition task (SignIn) code here. I'll appreciate your feedback.

Thank you

### CODE REVIEW

File: CompetionTaskMars/Tests/Hooks.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+            // Perform Login using credentials from JSON
+            var credentials = Utilities.JsonReader.GetCredentials(); 
+             signInObj.LoginActions(credentials.Email, credentials.Password);            
+            
+        }
+
+        [Test]
+        public void TestAddNewEducation()
+        {
```
Comment (amanbansalnz) [inline CompetionTaskMars/Tests/Hooks.cs:76]: i would move the tests into a there own skill tests etc classes

File: CompetionTaskMars/Tests/Hooks.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            var credentials = Utilities.JsonReader.GetCredentials();
+            test.Info("Starting 'Add New Education' test");
+
```
Comment (amanbansalnz) [inline CompetionTaskMars/Tests/Hooks.cs]: can you pass the files from the test, and each test should have its own json file

File: CompetionTaskMars/Tests/Hooks.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            educationActions.NavigateToEducationTab();
+            educationActions.AddNewEducationButtonActions();
+            educationActions.AddCollegeUniversityNameActions(credentials.College);
+            educationActions.AddCountyOfCollegeUniversityActions(credentials.Country);
```
Comment (amanbansalnz) [inline CompetionTaskMars/Tests/Hooks.cs]: this logic either create a step class eg skillsteps which has the method to add education and then you in the test you can just do skillSteps.addEducation or you can just move this to the pages and then in test do educationAction.addEducation

File: CompetionTaskMars/Tests/Hooks.cs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        [OneTimeSetUp]
+        public void SetupReport()
+        {
+            var htmlReporter = new ExtentSparkReporter("C:\\repo\\CompetionTaskMars\\Reports\\ExtentReport.html");
+            extent = new ExtentReports();
+            extent.AttachReporter(htmlReporter);
+            htmlReporter.Config.DocumentTitle = "Mars Competition Test Report";
+            htmlReporter.Config.ReportName = "Automation Test Report";
+            htmlReporter.Config.Theme = AventStack.ExtentReports.Reporter.Config.Theme.Standard;
+            extent.AddSystemInfo("UserName", "Kalpana Dissanayake");
```
Comment (amanbansalnz) [inline CompetionTaskMars/Tests/Hooks.cs]: remember to think about clean up

File: CompetionTaskMars/Utilities/JsonReader.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            var jsonFilePath = "C:\\repo\\CompetionTaskMars\\CompetionTaskMars\\CompetionTaskMars\\TestData\\testData.json"; 
+            var jsonData = File.ReadAllText(jsonFilePath);
+            return JsonConvert.DeserializeObject<Credentials>(jsonData);
+        }
+    }
+
+    public class Credentials
+    {
+        public string Email { get; set; }
```
Comment (amanbansalnz) [inline CompetionTaskMars/Utilities/JsonReader.cs]: would create new class for objects as separate files, maybe create a package called model or pojo and then create Credentials class and any other class that you need there

**Comment by kalpanaTech** [comment]
Hi Aman,  
Please find the "Competition task" automation code for the review. I'm looking forward to more advices to improve my code. I'll appreciate your feedback.  
Thanks

**Review by amanbansalnz — APPROVED** [review]
Hey Kalpana I am happy to approve your task after making the suggested changes. Just One thing to take note on is that make sure you do cleanup after each test run and also the cleanup for all data for that feature should happen in one time setup.

**Comment by kalpanaTech** [comment]
Hi Aman, I have pushed the changes you mentioned. I'm looking forward to more advices to improve my code. 
Thank you

**Pull request merged**